### DISCUSSION TITLE
Which node will run a SCRIPT in cluster mode with multiple shards?

### DISCUSSION THREAD

**Discussion by bibicu123** [redis/go-redis#2627] [Q&A]
I am trying to understand how go-redis will run a script with cluster enabled  and multiple shards?
Which node will choose to run the script on ?

Let's assume a script that reads a key A and writes another key B based on the value of A. To keep things simple, both keys hash to the same slot.

Is go-redis smart enough to parse the keys and run it on corresponding node ?

**Comment by elena-kolevska** [comment]
Yes, go-redis will choose a shard based on the first key in the argument list. 
https://github.com/redis/go-redis/blob/master/commands.go#L3402C3-L3402C3
https://github.com/redis/go-redis/blob/master/cluster.go#L1774
